aviary/mission/flops_based/phases/two_dof_phase.py

```diff
@@ -0,0 +1,267 @@
+
+        return phase
+
+    def make_default_transcription(self):
+        '''
+        Return a transcription object to be used by default in build_phase.
+        '''
+        user_options = self.user_options
+
+        num_segments, _ = user_options.get_item('num_segments')
+        order, _ = user_options.get_item('order')
+
+        seg_ends, _ = dm.utils.lgl.lgl(num_segments + 1)
+
+        transcription = dm.Radau(
+            num_segments=num_segments, order=order, compressed=True,
+            segment_ends=seg_ends)
+
+        return transcription
+
+    def _extra_ode_init_kwargs(self):
+        """
+        Return extra kwargs required for initializing the ODE.
+        """
+        # TODO: support external_subsystems and meta_data in the base class
+        return {
+            'external_subsystems': self.external_subsystems,
+            'meta_data': self.meta_data,
+            'subsystem_options': self.subsystem_options,
+            'input_speed_type': SpeedType.MACH,
+            'clean': self.user_options.get_val('clean'),
+            'ground_roll': self.user_options.get_val('ground_roll'),
+            'balance_throttle': self.user_options.get_val('balance_throttle'),
+        }
+
+
+TwoDOFPhase._add_meta_data(
+    'num_segments', val=5, desc='transcription: number of segments')
+
+TwoDOFPhase._add_meta_data(
+    'order', val=3,
+    desc='transcription: order of the state transcription; the order of the control'
+    ' transcription is `order - 1`')
+
+TwoDOFPhase._add_meta_data('fix_initial', val=False)
+
+TwoDOFPhase._add_meta_data('fix_duration', val=False)
+
+TwoDOFPhase._add_meta_data('optimize_mach', val=False)
+
+TwoDOFPhase._add_meta_data('optimize_altitude', val=False)
+
+TwoDOFPhase._add_meta_data('throttle_setting', val=None, desc='throttle setting')
+TwoDOFPhase._add_meta_data('initial_bounds', val=(0., 100.),
+                           units='s', desc='initial bounds')
+TwoDOFPhase._add_meta_data('duration_bounds', val=(
+    0., 3600.), units='s', desc='duration bounds')
+TwoDOFPhase._add_meta_data('initial_ref', val=100., units='s', desc='initial reference')
+TwoDOFPhase._add_meta_data('duration_ref', val=1000.,
+                           units='s', desc='duration reference')
+TwoDOFPhase._add_meta_data('control_order', val=1, desc='control order')
+TwoDOFPhase._add_meta_data('opt', val=True, desc='opt')
+TwoDOFPhase._add_meta_data('ground_roll', val=True)
+TwoDOFPhase._add_meta_data('rotation', val=False)
+TwoDOFPhase._add_meta_data('clean', val=False)
+TwoDOFPhase._add_meta_data('balance_throttle', val=False)
+TwoDOFPhase._add_meta_data('constraints', val={})
+TwoDOFPhase._add_meta_data('mach_bounds', val=(0., 2.), units='unitless')
+TwoDOFPhase._add_meta_data('altitude_bounds', val=(0., 60.e3), units='ft')
+
+TwoDOFPhase._add_initial_guess_meta_data(
+    InitialGuessTime(key='distance'),
+    desc='initial guess for initial time and duration specified as a tuple')
```

Comment:
time/distance?

Reply:
Not quite, here the "time" is distance as we are integrating with respect to distance.

Reply:
should we change the name of InitialGuessTime to reflect that it is the integration variable, not necessarily time?

Reply:
Good idea, changed!